qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -242,10 +242,8 @@ def run(self, dag):
                     qubit = run[0].qargs
                     for gate, angles in best_circuit_sequence:
                         op = CircuitInstruction.from_standard(gate, qubit, angles)
-                        node = DAGOpNode.from_instruction(op, dag=dag)
-                        node._node_id = dag._multi_graph.add_node(node)
-                        dag._increment_op(gate.name)
-                        dag._multi_graph.insert_node_on_in_edges(node._node_id, first_node_id)
+                        node = DAGOpNode.from_instruction(op)
+                        dag._insert_1q_on_incoming_qubit(node, first_node_id)
```

Comment:
This is a pretty niche optimization that really only applies to this case where we're contracting a sequence of 1q gates like this. I don't think it's generally useful outside of this, and we can just use `contract_nodes` or `substitute_node_with_dag`. Just exposing `insert_node_on_in_edges` is a bit risky because it adds the gate on all wires regardless of what they are which outside of the constraints where this path mutates the graph.

That being said I'm fine making this public if you think it's worthwhile we can just document it very explicitly as only support 1q gate without any classical wires. Although the usage here is going to be short lived as migrating `Optimize1qGatesDecomposition` is my first todo after this merges and this dag manipulation will move to rust space.

Reply:
Let's just leave it as-is then, but delete the private method during the migration.

Reply:
Just an FYI, I started a very rough draft of what this will look like here: https://github.com/mtreinish/qiskit-core/commit/2ffe1a1948d28cc6f1d09aca87440c8d08114593 it's not tuned and pretty rough still. But it's working for 80% of the pass's tests.

Reply:
#12959 will remove this function.